### ISSUE TITLE
Wrong storage when selecting external storage

### ISSUE THREAD

**Issue by red-yellow** [avluis/Hentoid#411]
**Hentoid Version**: 1.9.3 (157)

**Device/Android Version**: Amazon Fire 5th (KDDOWI) / Android 5.1.1

**Issue details / Repro steps**: The Kindle has a MicroSD which I am trying to target for Hentoid downloads. After selecting 'Select Downloads Folder' and the file chooser pops up, I choose the microsd card from the sidebar. However, the app saves to the internal storage instead. 

More specifically, it replaces the root device in the file path. So when I want to choose `/storage/sdcard1/h` to save, it instead saves to `/storage/emulated/0/h`. I also tried to select `/storage/sdcard1/Android/data/me.devsaki.hentoid/` but it saved to `/storage/emulated/0/Android/data/me.devsaki.hentoid/`. See attached images.

**Stack trace / LogCat**:
Selecting storage/emulated/0/h/:
```ruby
I/Vlog    (  916): performance:SkippedVsyncsBucket3:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=1.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:SkippedVsyncsBucket5:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=1.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:FrameDropTotal:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=16.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:FramesTotal:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=56.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:DelayedFirstFrames:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=1.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:FirstFramesTotal:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=7.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Kernel  ( 6242): [105635.690029]-(0)[6242:oid.documentsui]dpi vsync
D/Documents( 6242): Current stack:
D/Documents( 6242):  * Root{authority=com.android.externalstorage.documents, rootId=primary, title=Internal storage}
D/Documents( 6242):  +-- Document{docId=primary:h, name=h}
D/Documents( 6242):  +-- Document{docId=primary:, name=0}
D/Documents( 6242): userMode=0, userSortOrder=0 --> mode=1, sortOrder=1
D/Documents( 6242): onFinished() [content://com.android.externalstorage.documents/tree/primary%3Ah]
W/EMP.LoggerMetricsFactoryImpl(  916): EMP.MetricsCollectionAgent:AppUsageReporter:appVersionName=5.1.1-37.6.2.6_user_626533320;DV;1,appCollectionReason=platformKey;DV;1,appVersionCode=22;DV;1,Product_Mode=DEFAULT;DV;1,appUsageCollection=required;DV;1,durationMS=6872;DV;1,startTimeEpoch=1572997340622;DV;1,launchCount=1;DV;1,appPackageName=com.android.documentsui;DV;1,page-type=AppUsageStats;CK;1,hitType=appAction;CK;1,team-name=KindleFramework;CK;1,site_variant=KindleApplication;CK;1,page-action=DISPLAY_USAGE;CK;1,RequestId=8140B348F0914401C01F;DV;1:NA
V/WindowManager(  916): Adding window Window{f53c4e5 u0 me.devsaki.hentoid/me.devsaki.hentoid.activities.ImportActivity} at 11 of 18 (before Window{6215262 u0 com.android.documentsui/com.android.documentsui.DocumentsActivity})
I/Vlog    (  916): Thermal:Screen_On_Thermal_Throttling:fgtracking=false;DV;1,key=0;DV;1,Timer=0.0018869444444444443;TI;1,unit=hours;DV;1,metadata=com.android.documentsui;DV;1:HI
I/Vlog    (  916): ScreenTime:PackageScreenTime:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Timer=0.0018869444444444443;TI;1,unit=hours;DV;1,metadata=!{"d"#{"app_version"#"22"}};DV;1:HI
I/LoggerMetricsFactoryImpl(  916): PhoneWindowManager:ScreenTime:endTimeMs=1.572997347607E12;TI;1,startTimeMs=1.572997340814E12;TI;1,com.android.documentsui=6793.0;TI;1:NR
I/Vlog    (  916): performance:warm_app_warm_transition_launch_time:fgtracking=false;DV;1,key=me.devsaki.hentoid;DV;1,Timer=114.0;TI;1,unit=ms;DV;1,metadata=me.devsaki.hentoid/.activities.ImportActivity!{"d"#{"app_version"#"157"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"com.android.documentsui/.DocumentsActivity"}};DV;1:HI
I/Vlog    (  916): performance:warm_activity_launch_time:fgtracking=false;DV;1,key=me.devsaki.hentoid;DV;1,Timer=114.0;TI;1,unit=ms;DV;1,metadata=me.devsaki.hentoid/.activities.ImportActivity!{"d"#{"app_version"#"157"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"com.android.documentsui/.DocumentsActivity"}};DV;1:HI
I/Vlog    (  916): performance:SkippedVsyncsBucket1:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=3.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:SkippedVsyncsBucket3:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=1.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:FrameDropTotal:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=7.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:FramesTotal:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=69.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:FirstFramesTotal:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=3.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
```

Selecting /storage/sdcard1/h/:
```ruby
I/Vlog    (  916): performance:SkippedVsyncsBucket5:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=1.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:FrameDropTotal:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=13.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:FramesTotal:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=34.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:DelayedFirstFrames:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=1.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
I/Vlog    (  916): performance:FirstFramesTotal:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Counter=4.0;CT;1,unit=count;DV;1,metadata=com.android.documentsui/.DocumentsActivity!{"d"#{"app_version"#"22"}$"m"#{"context"#"S1T0A1D0I0L0E0M99"$"prev"#"me.devsaki.hentoid/.activities.ImportActivity"}};DV;1:HI
D/Documents( 6242): Current stack:
D/Documents( 6242):  * Root{authority=com.android.externalstorage.documents, rootId=2428-6EF4, title=android}
D/Documents( 6242):  +-- Document{docId=2428-6EF4:h, name=h}
D/Documents( 6242):  +-- Document{docId=2428-6EF4:, name=sdcard1}
D/Documents( 6242): userMode=0, userSortOrder=0 --> mode=1, sortOrder=1
D/Documents( 6242): onFinished() [content://com.android.externalstorage.documents/tree/2428-6EF4%3Ah]
W/EMP.LoggerMetricsFactoryImpl(  916): EMP.MetricsCollectionAgent:AppUsageReporter:appVersionName=5.1.1-37.6.2.6_user_626533320;DV;1,appCollectionReason=platformKey;DV;1,appVersionCode=22;DV;1,Product_Mode=DEFAULT;DV;1,appUsageCollection=required;DV;1,durationMS=4712;DV;1,startTimeEpoch=1572997467230;DV;1,launchCount=1;DV;1,appPackageName=com.android.documentsui;DV;1,page-type=AppUsageStats;CK;1,hitType=appAction;CK;1,team-name=KindleFramework;CK;1,site_variant=KindleApplication;CK;1,page-action=DISPLAY_USAGE;CK;1,RequestId=2041361347D4FCF86F8D;DV;1:NA
V/WindowManager(  916): Adding window Window{b6677b3 u0 me.devsaki.hentoid/me.devsaki.hentoid.activities.ImportActivity} at 11 of 18 (before Window{7d4c8b1 u0 com.android.documentsui/com.android.documentsui.DocumentsActivity})
I/Vlog    (  916): Thermal:Screen_On_Thermal_Throttling:fgtracking=false;DV;1,key=0;DV;1,Timer=0.0012822222222222221;TI;1,unit=hours;DV;1,metadata=com.android.documentsui;DV;1:HI
I/Vlog    (  916): ScreenTime:PackageScreenTime:fgtracking=false;DV;1,key=com.android.documentsui;DV;1,Timer=0.0012822222222222221;TI;1,unit=hours;DV;1,metadata=!{"d"#{"app_version"#"22"}};DV;1:HI
I/LoggerMetricsFactoryImpl(  916): PhoneWindowManager:ScreenTime:endTimeMs=1.572997472055E12;TI;1,startTimeMs=1.572997467439E12;TI;1,com.android.documentsui=4616.0;TI;1:NR
```

![external-dir](https://user-images.githubusercontent.com/57420557/68255309-c0454380-ffe1-11e9-8f45-648c894b4aa1.png)
![filesystems](https://user-images.githubusercontent.com/57420557/68255310-c0454380-ffe1-11e9-84d8-9cfbaad06865.png)
![it-saved-to-internal](https://user-images.githubusercontent.com/57420557/68255311-c0ddda00-ffe1-11e9-9c1c-e42d63ace2f4.png)
![its-empty](https://user-images.githubusercontent.com/57420557/68255313-c0ddda00-ffe1-11e9-9e1b-1b852b80d24c.png)

@AVnetWS/admin-team

**Comment by red-yellow** [comment]
P.S. This was happening on the previous releases as well; it's not because of the update.

**Comment by RobbWatershed** [comment]
Thanks a lot for the very detailed report.
This bug has been identified and solved (see #405). The fix will be available on the next beta.
